visma/matrix/structure.py

```diff
@@ -103,22 +131,37 @@ def determinant(self):
         pass
 
 
-class IdenMat(SquareMat):
-    """Class for identity matrix
+class DiagMat(SquareMat):
+    """Class for Diagonal matrix
 
-    Identity matrix is a square matrix with all elements as 0 except for the diagonal elements which are 1.
+    Diagonal matrix is a square matrix with all elements as 0 except for the diagonal elements.
 
     Extends:
         SquareMat
     """
 
-    def __init__(self, dim):
+    def __init__(self, dim, diagElem):
         super().__init__()
         for i in range(0, dim[0]):
             row = []
             for j in range(0, dim[1]):
                 if i == j:
-                    row.append(Constant(1))
+                    row.append(Constant(diagElem[i]))
                 else:
                     row.append(Constant(0))
```

Comment:
The `IdenMat` class I wrote had some faults
```suggestion
                     row.append([Constant(0)])
```